src/Uno.UI/UI/Xaml/UIElement.cs

```diff
@@ -534,10 +536,18 @@ internal static Matrix3x2 GetTransform(UIElement from, UIElement to)
 				// Unfortunately we didn't find the 'to' in the parent hierarchy,
 				// so matrix == fromToRoot and we now have to compute the transform 'toToRoot'.
 				// Note: We do not propagate the 'intermediatesSelector' as cached transforms would be irrelevant
-				var toToRoot = GetTransform(to, null);
-				var rootToTo = toToRoot.Inverse();
+				var toContext = new TransformToVisualContext();
+				var toToRoot = GetTransform(to, null, toContext);
 
-				matrix *= rootToTo;
+#if __IOS__
+				// On iOS, the `from` and `to` may be coming from different ViewController.
+				// In such case, their coordinates should not be "added" together, since they are from different coordinates space.
+				if (context.ViewController == toContext.ViewController)
```

Comment:
@Xiaoy312 there are valid situations where the `ViewController` is null.

Reply:
i guess we can also the top-most crawled root in the context,
and compare if they are equal when both vc are null, to decide whether the offsets/matrices should be aggregated or not

Reply:
You mean finding their respective roots and making sure they are part of the same tree? My only concern is figuring out how expensive this can get.

Reply:
we are already traversing the visual-tree, just return the last known element from TryGetParentUIElementForTransformToVisual before we return

Reply:
As we discuss on Teams, we will check this as an edge case and it will be worked under a different GH issue https://github.com/unoplatform/uno/issues/12250